tests/acceptance/01_vars/02_functions/variablesmatching.cf

```diff
@@ -21,25 +25,15 @@ bundle agent test
       "x_vars" slist => variablesmatching("default:init.test_fbeae67f3e347b5e0032302200141131.*", "x");
       "z_vars" slist => variablesmatching("default:init.test_fbeae67f3e347b5e0032302200141131.*", "z");
 
-      "count" int => length(vars);
-      "x_count" int => length(x_vars);
-      "z_count" int => length(z_vars);
+      "fullvars" data => data_matching("default:init.test_b37ce1b03955522848f0a82d0b2b2e21.*");
```

Comment:
Sorry, this was just badly phrased. The function finds variables whose name matches the argument, just like `variablesmatching()`. But instead of returning a list of names, it returns a key-value map with those names as keys, and the variable contents as values.